### PR TITLE
[detours] Fix to build with debug/release and static/dynamic crt, and cmake target

### PR DESCRIPTION
**Pull request by ivysnow**
allow build detours with debug/release config, use static/dynamic crt linkage.
add cmake target for detours.

### CODE REVIEW

File: ports/detours/portfile.cmake [hunk 1]
```diff
@@ -10,18 +10,35 @@ vcpkg_from_github(
...
+
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/unofficial-detours TARGET_PATH share/unofficial-detours)
+vcpkg_copy_pdbs()
+
+configure_file(
+    ${CMAKE_CURRENT_LIST_DIR}/unofficial-detours-config.in.cmake
+    ${CURRENT_PACKAGES_DIR}/share/unofficial-detours/unofficial-detours-config.cmake
+    @ONLY
 )
```
Comment (PhoebeHui) [inline ports/detours/portfile.cmake]: Could you move this to CMakeList.txt file?

File: ports/detours/portfile.cmake [hunk 2]
```diff
@@ -10,18 +10,35 @@ vcpkg_from_github(
...
 
 vcpkg_build_nmake(
     SOURCE_PATH ${SOURCE_PATH}
     PROJECT_SUBPATH "src"
     PROJECT_NAME "Makefile"
     OPTIONS "PROCESSOR_ARCHITECTURE=${VCPKG_TARGET_ARCHITECTURE}"
-    NO_DEBUG
 )
```
Comment (PhoebeHui) [inline ports/detours/portfile.cmake]: Since we have CMakeList.txt file here, we suggest to remove this and writing everything to CMakeList.txt file.

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/detours/portfile.cmake (hunk 2 above)
Comment (ivysnow) [reply ports/detours/portfile.cmake]: Ok I'll try it.

**Comment by PhoebeHui** [comment]
@ivysnow, do you have a chance to address the review suggestions? Thanks!

**Comment by LilyWangL** [comment]
Pinging @ivysnow for response. Is work still being done for this PR?

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**